packages/tpetra/core/src/Tpetra_Details_unpackCrsMatrixAndCombine_def.hpp

```diff
@@ -529,6 +818,53 @@ compute_total_num_entries (
   return tot_num_ent;
 }
 
+template<class LO>
+KOKKOS_INLINE_FUNCTION
+size_t
+unpackRowCount(const char imports[],
+               const size_t offset,
+               const size_t num_bytes)
+{
+  using PT = PackTraits<LO>;
+
+  LO num_ent_LO = 0;
+  if (num_bytes > 0) {
+    const size_t p_num_bytes = PT::packValueCount(num_ent_LO);
+    if (p_num_bytes > num_bytes) {
+      return OrdinalTraits<size_t>::invalid();
+    }
+    const char* const in_buf = imports + offset;
+    (void) PT::unpackValue(num_ent_LO, in_buf);
+  }
+  return static_cast<size_t>(num_ent_LO);
+}
+
+/// \brief Compute the index and batch number associated with each batch
+///
+/// batch_info(i, 0) is the local index of the ith batch
+/// batch_info(i, 1) is the local batch number of the ith batch
+template<class View1, class View2>
+KOKKOS_INLINE_FUNCTION
+bool
```

Comment:
This looke like a host only function. Then, you can replace KOKKOS_INLINE_FUNCTION with inline.